e2e/nomostest/artifactregistry/image.go

```diff
@@ -248,14 +255,12 @@ func generateImageName(nt *nomostest.NT, chartName string) string {
 	return chartName
 }
 
-// generateImageVersion returns the version with a random 8 character suffix.
-// Result will be no more than 20 characters and can function as a k8s metadata.name.
+// validateImageVersion will validate if chart version string
+// is 20 characters or less and can function as a k8s metadata.name.
 // Chart name and version must be less than 63 characters combined.
-func generateImageVersion(chartVersion string) string {
-	if len(chartVersion) > 12 {
-		chartVersion = chartVersion[:12]
-		chartVersion = strings.Trim(strings.Trim(chartVersion, "-"), ".")
+func validateImageVersion(chartVersion string) (string, error) {
+	if len(chartVersion) > 20 {
+		return chartVersion, fmt.Errorf("chart version string %s should not exceed 20 characters", chartVersion)
```

Comment:
Use %q for the value so it's easy to tell if it's empty or has spaces.

Reply:
replaced with %q